src/vm.rs

```diff
@@ -262,14 +262,14 @@ impl<'a, E: UserDefinedError> EbpfVm<'a, E> {
     /// // Register a syscall.
     /// // On running the program this syscall will print the content of registers r3, r4 and r5 to
     /// // standard output.
-    /// vm.register_syscall(6, bpf_trace_printf::<UserError>).unwrap();
+    /// vm.register_syscall(6, Syscall::Function(bpf_trace_printf::<UserError>)).unwrap();
     /// ```
     pub fn register_syscall(
```

Comment:
I suppose this is used in tests but maybe we don't need it as part of the public api any longer